Tests/IntegrationTests/FastEndpoints.IntegrationTests/WebTests/MiscTestCases.cs

```diff
@@ -290,6 +290,221 @@ public async Task DupeParamBindingToIEnumerableProps()
+            Child = new()
+            {
+                Age = -22,
+                Name = "child name",
+                Id = Guid.Parse("8bedccb3-ff93-47a2-9fc4-b558cae41a06"),
+                Strings = new()
+                {
+                    "string1", "string2", "", "strangeString"
+                }
+            },
+            Numbers = new() { 0, 1, -222, 1000, 22 }
+        });
+    }
+
+    [Fact]
+    public async Task BindingObjectFromQueryUse()
+    {
+        var (rsp, res) = await GuestClient
+            .GETAsync<TestCases.QueryObjectBindingTest.Request, TestCases.QueryObjectBindingTest.Response>(
+                "api/test-cases/query-object-binding-test" +
+                "?BoOl=TRUE&String=everything&iNt=99&long=483752874564876&DOUBLE=2232.12&Enum=3" +
+                "&age=45&name=john&id=10c225a6-9195-4596-92f5-c1234cee4de7" +
+                "&numbers[0]=0&numbers[1]=1&numbers[2]=-222&numbers[3]=1000&numbers[4]=22" +
+                "&favoriteDay=Friday&IsHidden=FALSE&ByteEnum=2" +
+                "&child.id=8bedccb3-ff93-47a2-9fc4-b558cae41a06" +
+                "&child.name=child name&child.age=-22" +
+                "&CHILD.FavoriteDays[0]=1&ChiLD.FavoriteDays[1]=Saturday&CHILD.ISHiddeN=TruE" +
+                "&child.strings[0]=string1&child.strings[1]=string2&child.strings[2]=&child.strings[3]=strangeString",
+                new()
+                {
+                });
+
+        rsp?.StatusCode.Should().Be(HttpStatusCode.OK);
+        res?.Should().BeEquivalentTo(
+            new TestCases.QueryObjectBindingTest.Request
+            {
+                Double = 2232.12,
+                Bool = true,
+                Enum = DayOfWeek.Wednesday,
+                String = "everything",
+                Int = 99,
+                Long = 483752874564876,
+                Person = new()
+                {
+                    Age = 45,
+                    Name = "john",
+                    Id = Guid.Parse("10c225a6-9195-4596-92f5-c1234cee4de7"),
+                    FavoriteDay = DayOfWeek.Friday,
+                    ByteEnum = TestCases.QueryObjectBindingTest.ByteEnum.AnotherCheck,
+                    IsHidden = false,
+                    Child = new()
+                    {
+                        Age = -22,
+                        Name = "child name",
+                        Id = Guid.Parse("8bedccb3-ff93-47a2-9fc4-b558cae41a06"),
+                        Strings = new()
+                        {
+                            "string1", "string2", "", "strangeString"
+                        },
+                        FavoriteDays = new() { DayOfWeek.Monday, DayOfWeek.Saturday },
+                        IsHidden = true
+                    },
+                    Numbers = new() { 0, 1, -222, 1000, 22 }
+                }
+            }
+            );
+    }
+
+
+    [Fact]
+    public async Task BindingArraysOfObjectsFromQueryUse()

```

Comment:
this test is currently not passing. this is still not done right?

Reply:
Yep, I'll try to finish this part today

Reply:
okie then

Reply:
Mb I'll rewrite some parts to use read-only spans instead of strings